src/NzbDrone.Core/Parser/LanguageParser.cs

```diff
@@ -226,6 +231,56 @@ public static Language ParseSubtitleLanguage(string fileName)
             return Language.Unknown;
         }
 
+        public static (List<string> languageTags, string title) ParseLanguageTagsAndTitle(string fileName, Episode episode)
+        {
+            var simpleFilename = Path.GetFileNameWithoutExtension(fileName);
+            var matchTitle = SubtitleLanguageTitleRegex.Match(simpleFilename);
+
+            if (matchTitle.Groups["iso_code"].Captures.Count != 1)
+            {
+                throw new ArgumentException("Subtitle file title probably parsed incorrectly, not using.");
+            }
+
+            var languageTags = matchTitle.Groups["tags1"].Captures
+                .Union(matchTitle.Groups["tags2"].Captures)
+                .Cast<Capture>()
+                .Where(tag => !tag.Value.Empty())
+                .Select(tag => tag.Value.ToLower());
+            var title = matchTitle.Groups["title"].Captures.Cast<Capture>().First().ToString();
+
+            if (matchTitle.Groups["tags1"].Captures.Empty())
+            {
+                var episodeFile = episode.EpisodeFile.Value;
+                var episodeFileTitle = Path.GetFileNameWithoutExtension(episodeFile.RelativePath);
+                var originalEpisodeFileTitle = Path.GetFileNameWithoutExtension(episodeFile.OriginalFilePath) ?? string.Empty;
+
+                if (episodeFileTitle.Contains(title, StringComparison.OrdinalIgnoreCase) || originalEpisodeFileTitle.Contains(title, StringComparison.OrdinalIgnoreCase))
```

Comment:
This logic shouldn't live in the parser, like language parsing this should probably live in a augmenter.

Reply:
In so far as I understand the augmenters exist to add the rest of the available info to a partial local episode where possible. What exactly should I be augmenting here? Should I create a new set of augmenters that add info such as title, copies etc.? Those would be entirely separate from the existing local episode augmenters tho.

Reply:
They do several things, but I'm specifically referring to this: https://github.com/Sonarr/Sonarr/blob/develop/src/NzbDrone.Core/MediaFiles/EpisodeImport/Aggregation/Aggregators/Augmenters/Language/AugmentLanguageFromFileName.cs#L22-L27

The parser should only parse out information, it shouldn't try to make sense of what it parsed out and strip information. Instead that can be stripped out later.

Reply:
The example you gave in the other comment is why I'm actually checking both the original and current file name for matches. I guess I could look at the entire history of renames of the file(somehow, need to see how that might be possible)? Or at least at the previous name.

Reply:
What I still find strange is how augmenters typically operate on local episodes-this would essentially mean having the list of the subtitle files as part of the local episode and then just import that I suppose?